js/src/Ice/InstanceExtensions.js

```diff
@@ -181,9 +183,19 @@ Instance.prototype.finishSetup = function (communicator) {
             this._initData.properties.getIcePropertyAsInt("Ice.Connection.Client.InactivityTimeout"),
         );
 
+        const programName = this._initData.properties.getIceProperty("Ice.ProgramName");
+        const logFile = this._initData.properties.getIceProperty("Ice.LogFile");
         if (this._initData.logger === null) {
-            this._initData.logger = getProcessLogger();
+            if (logFile.length > 0) {
+                if (FileLogger === null) {
+                    throw new InitializationException("Ice.LogFile property is not supported in Web Browsers");
+                }
+                this._initData.logger = new FileLogger(programName, logFile);
+            } else {
+                this._initData.logger = new Logger(programName);
+            }
         }
+        setProcessLogger(this._initData.logger);
```

Comment:
I don't understand this change. The process logger should not be the logger of the most recently created communicator.

Reply:
I’d like to avoid lazy initialization in ProcessLogger, since it complicates setting the prefix for the default logger. What about:

```
if (getProcessLogger() === null) {
    setProcessLogger(this._initData.logger);
}
```

Reply:
The process logger and initData.logger are/should be totally independent.